workbench-core/authentication/src/plugins/cognitoAuthenticationPlugin.ts

```diff
@@ -29,45 +29,59 @@ interface TokensExpiration {
   refreshToken: number; // ms
 }
 
-export interface CognitoAuthenticationPluginOptions {
+interface CognitoAppClient {
   /**
-   * The Cognito domain. Follows the format: "https://\<domain prefix\>.auth.\<region\>.amazoncognito.com"
+   * The Cognito app client IDs.
    */
-  cognitoDomain: string;
-
+  clientId: string | string[];
   /**
    * The Cognito user pool ID. Follows the format: "\<region\>_\<some string\>"
    */
   userPoolId: string;
```

Comment:
Every AppClient should belong to the same user pool

Reply:
Ticket describes two scenarios, one is passing multiple AppClients from same user pool but second part describes passing various user pool AppClients:

```
Ask for from 
[@thingut](https://amzn-aws.slack.com/team/W017VKKSHS5)
: We need to provide a way for access tokens generated from other user pool clients to pass the verifyToken method.
```

Reply:
The wording of the ask is a little confusing, but they are asking for multiple app clients from the same user pool.

Reply:
The decision to allow users from other user pools authenticate with this service requires further discussion and is outside the scope of this ticket.